packages/teleport/src/components/BannerList/useAlerts.tsx

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { useState, useEffect } from 'react';
+import Logger from 'shared/libs/logger';
+
+import { fetchClusterAlerts } from 'teleport/services/alerts';
+import useStickyClusterId from 'teleport/useStickyClusterId';
+
+import type { ClusterAlert } from 'teleport/services/alerts';
+
+const logger = Logger.create('ClusterAlerts');
+
+const DISABLED_BANNERS = 'disabledAlerts';
+
+function addHours(date: number, hours: number) {
+  return date + hours * 60 * 60 * 1000;
+}
+
+function getItem(key: string): string | null {
+  return window.localStorage.getItem(key);
+}
+
+function setItem(key: string, data: string) {
+  window.localStorage.setItem(key, data);
+}
+
+type DismissedAlert = {
+  [alertName: string]: string;
+};
+
+export function useAlerts() {
+  const [alerts, setAlerts] = useState<ClusterAlert[]>([]);
+  const [dismissedAlerts, setDismissedAlerts] = useState<DismissedAlert[]>([]);
+  const { clusterId } = useStickyClusterId();
+
+  useEffect(() => {
+    const disabledAlerts = getItem(DISABLED_BANNERS);
+    if (disabledAlerts) {
+      // Loop through the existing ones and remove those that have passed 24h.
+      const data = JSON.parse(disabledAlerts);
+      Object.entries(data).forEach(([name, expiry]) => {
+        if (new Date().getTime() > expiry) {
```

Comment:
Comparing string(eg `'2022-09-09T07:53:09.157Z'`) and number(eg `1662709995148`) won't work - this will always produce `false`. The reason why `ts` wasn't complaining about this line is that expiry has implicit any type (we plan to address that in https://github.com/gravitational/webapps/issues/957)
```suggestion
        if (new Date().getTime() > Date.parse(expiry)) {
```

Reply:
I think we should add test case to cover that

Reply:
good catch! I actually had this originally as ` if (new Date().getTime() > +expiry) {` but the + was dropped somewhere along the line.

I was actually in process of writing tests for this just didn't get it finished by EOD yesterday so I'll be sure that there is a test case for this.

Reply:
```js
+'2022-09-09T07:53:09.157Z'
``` 
won't work since it will produce `NaN`

Reply:
I'm not sure I understand? `expiry` is epoch time not a datestring.

Reply:
You are right. I thought we were sanitizing js dates (for unknown reason :) ). I think it should work as is now. Sorry for unnesessery discussion

Reply:
no problem at all :)